utils/docker/Dockerfile.centos.8

```diff
@@ -9,31 +9,63 @@
-      MY_REPO="${REPO_URL}repository/centos-${version}-x86_64-group/";         \
-      MY_NAME="${MY_REPO#*//}";						                           \
-      MY_NAME="${MY_NAME//\//_}";					                           \
+# If a local distro repository is supplied, then we should use it
+# and disable using the distro repositories that can be mirrored.
+# The Docker image starts out with the DISTRO provided GPG keys
+# on the disk, but not installed.  These must be isntalled before
+# adding any repo that needs GPG keys but does not provide meta
+# data for installing those keys.  Then the epel-release
+# package must be installed and the GPG key it provides must be
+# installed.
+RUN if [ -n "$REPO_DISTRO" ]; then                              \
+      rpm --import /etc/pki/rpm-gpg/*;                          \
+      MY_REPO="${REPO_URL}${REPO_DISTRO}/";                     \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
       echo -e "[${MY_NAME}]\n\
 name=created from ${MY_REPO}\n\
 baseurl=${MY_REPO}\n\
 enabled=1\n\
-priority=1\n\
-gpgcheck=False\n" >> /etc/yum.repos.d/local-centos-group.repo;                 \
+repo_gpgcheck=0\n\
+gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      dnf clean all;                                            \
+      dnf list --showduplicates epel-release dnf-plugins-core dnf-utils; \
+      dnf -y install --disablerepo extras --disablerepo baseos  \
+          epel-release-8-8.el8 dnf-plugins-core dnf-utils;      \
+      rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8;         \
+      dnf config-manager --assumeyes --quiet --disable          \
+          baseos powertools epel epel-modular extras;           \
+      sed "s/^mirrorlist_expire=0*/mirrorlist_expire=99999999/" \
+          -i /etc/dnf/dnf.conf;                                 \
+      dnf -y update epel-release;                               \
+      dnf clean all;                                            \
     fi
 
-# Install basic tools to build DAOS
+# If a local DAOS repository is supplied, then we should use it
+# This is mainly for quickbuilds
+RUN if [ -n "$REPO_DAOS" ]; then                                \
+      MY_REPO="${REPO_URL}${REPO_DAOS}/";                       \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
+      echo -e "[${MY_NAME}]\n\
+name=created from ${MY_REPO}\n\
+baseurl=${MY_REPO}\n\
+enabled=1\n\
+repo_gpgcheck=0\n\
+gpgcheck=0\n" >> /etc/yum.repos.d/local-daos-group.repo;        \
+    fi
 
-RUN dnf clean all && dnf -y install epel-release dnf-plugins-core && \
-    dnf config-manager --set-enabled powertools && \
+
+# Install basic tools to build DAOS
+# *** Keep these in as much alphbetical order as possible ***
+RUN dnf clean all && \
     dnf -y upgrade && \
     dnf -y install \
```

Comment:
epel-release is potentially not installed at this point (in fact it won't be if no args are given), is that expected?  Given that if it is installed it appears to be disabled, and our GA builds won't have it maybe we can drop epel here?